scripts/create-kubestellar-demo-env.sh

```diff
@@ -134,14 +97,85 @@ checking_cluster() {
             
         else
             echo "$1 not found. Trying again..."
-            sleep 5
+            sleep 20
         fi
 
     done
 }
+##########################################
+
+echo -e "\nStarting environment clean up..."
+echo -e "Starting cluster clean up..."
+
+yq -i 'del(.preferences)' ${KUBECONFIG:-$HOME/.kube/config} 
+
+cluster_clean_up "kind delete cluster --name kubeflex" &
+cluster_clean_up "kind delete cluster --name cluster1" &
+cluster_clean_up "kind delete cluster --name cluster2" &
+wait
+echo -e "Cluster space clean up has been completed"
+
+echo -e "\nStarting context clean up..."
+context_clean_up
+echo "Context space clean up completed"
+
+echo -e "\nStarting the process to install KubeStellar core: kind-kubeflex..."
+clusters=(cluster1 cluster2)
+for cluster in "${clusters[@]}"; do
+   (
+     echo -e "Creating cluster ${cluster}..."
+     kind create cluster --name "${cluster}" >/dev/null 2>&1 &&
+     kubectl config rename-context "kind-${cluster}" "${cluster}" >/dev/null 2>&1
+     echo -e "${cluster} creation and context setup complete"
+   ) &
+done
+wait 
+
+echo -e "Creating KubeFlex cluster with SSL Passthrough"
+curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v${kubestellar_version}/scripts/create-kind-cluster-with-SSL-passthrough.sh | bash -s -- --name kubeflex --port 9443 
+echo -e "Completed KubeFlex cluster with SSL Passthrough"
+
+echo -e "\nPulling container images local..."
+images=("ghcr.io/loft-sh/vcluster:0.16.4"
+        "rancher/k3s:v1.27.2-k3s1"
+        "quay.io/open-cluster-management/registration-operator:v0.13.2"
+        "docker.io/bitnami/postgresql:16.0.0-debian-11-r13")
+
+for image in "${images[@]}"; do
+    (
+        docker pull "$image" && kind load docker-image "$image" --name kubeflex
+    ) &
+done
+
+helm upgrade --install ks-core oci://ghcr.io/kubestellar/kubestellar/core-chart \
+    --version $kubestellar_version \
+    --set-json='ITSes=[{"name":"its1"}]' \
+    --set-json='WDSes=[{"name":"wds1"}]' \
+    --set-json='verbosity.default=5'
+
+kubectl config delete-context wds1 || true
```

Comment:
I made a modified version of this script with the `kubectl config delete-context wds1 || true` and `kubectl config delete-context its1 || true` commands deleted, and tested. Works fine. These commands are not needed because they are redundant with the context deletions done above in the `context_clean_up` function.

Reply:
I will test again with them removed, but I have had mixed results in the past. I know that it should not affect the run, but I suspect it has.

Reply:
looks like removing the commands worked on my latest run - will check in this update in a bit

Reply:
check in removal in latest commit